llvm/lib/TargetParser/Triple.cpp

```diff
@@ -933,6 +933,9 @@ static Triple::ObjectFormatType getDefaultFormat(const Triple &T) {
     case Triple::Win32:
     case Triple::UEFI:
       return Triple::COFF;
+    case Triple::UnknownOS:
+      return T.getVendor() == Triple::Apple ? Triple::MachO : Triple::ELF;
+      // Intentional leak into the default case for additional logic.
     default:
       return T.isOSDarwin() ? Triple::MachO : Triple::ELF;
```

Comment:
Should we fold the above check into this line?
```suggestion
      return T.isOSDarwin() || (T.getVendor() == Triple::Apple)  ? Triple::MachO : Triple::ELF;
```
Or do we really only want it when the OS is unknown?

Reply:
Off the top of my head: For the sake of future proof, e.g. say Apple ships their own Linux, I assume their Linux will use ELF format, and so for case where "vendor == Apple && os == Linux", we probably want to return ELF.